console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,90 @@ type ConsolePluginSpec struct {
 	// i18n is the configuration of plugin's localization resources.
 	// +optional
 	I18n ConsolePluginI18n `json:"i18n"`
+	// contentSecurityPolicy is a list of Content Security Policy directives for the plugin.
+	// Each directive specifies a list of values that indicate server origins and script endpoints
+	// from which the plugin's assets can be loaded. This helps guard against cross-site
+	// scripting (XSS) attacks by enforcing strict security policies for asset loading.
+	// Dynamic plugins should to specify this field if they are loading assets form outside
+	// the cluster or if violation reports are observed.
+	// CSP violation reports can be viewed in browser's console during development and testing
+	//  of the plugin in the OpenShift web console.
+	// Available directives are default-src, script-src, img-src, style-src and font-src.
+	// Each of the available CSP directive may be defined only once in the list.
+	// By default the console server adds the value 'self'to all the various 'src' directives.
+	// For more information about the CSP directives, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
+	//
+	// The OpenShift web console server aggregates the CSP directives and values across
+	// its own default values and all enabled ConsolePlugin CRs, merging them to set a unified
+	// CSP header.
+	//
+	// Example:
+	//   ConsolePlugin A directives:
+	//     script-src: https://script1.com/ https://script2.com/
```

Comment:
A string with multiple entries, space separated? Why not a list of strings? I think this is not rendered correctly and we should resolve.

Or are you representing the headers?

Reply:
this is just an example how to source value are being unified  by the console server. Will add a the final header example as well 👍